plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -281,18 +302,51 @@ object StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext<ResultTrack
         return UnitDomain.element
     }
 
+    @OptIn(DfaInternals::class)
     override fun visitVariableAssignment(
         variableAssignment: FirVariableAssignment,
         data: StmtConversionContext<ResultTrackingContext>,
     ): Exp {
+
         // It is not entirely clear whether we can get away with ignoring the distinction between
         // lvalues and rvalues, but let's try to at first, and we'll fix it later if it turns out
         // not to work.
-        val convertedLValue = data.convert(variableAssignment.lValue)
-        val lValueType = data.embedType(variableAssignment.lValue)
-        val convertedRValue = data.convert(variableAssignment.rValue)
-        val rValueType = data.embedType(variableAssignment.rValue)
-        data.addStatement(Stmt.assign(convertedLValue, convertedRValue.convertType(rValueType, lValueType)))
+
+        val isClassField = if (variableAssignment.lValue is FirPropertyAccessExpression) {
+            (variableAssignment.lValue as FirPropertyAccessExpression).dispatchReceiver !is FirNoReceiverExpression
+        } else {
+            false
+        }
+
+        if (isClassField) {
+
+            val lValue = variableAssignment.lValue as FirPropertyAccessExpression
+            val lValueType = data.embedType(variableAssignment.lValue)
+            val receiver = data.convert(lValue.dispatchReceiver)
+
+            val symbol = lValue.symbol!! as FirPropertySymbol
+
+            val varEmbedding = VariableEmbedding(symbol.callableId.embedName(), lValueType)
+            val fieldAccess = Exp.FieldAccess(receiver, varEmbedding.toField())
+
+            val convertedRValue = data.convert(variableAssignment.rValue)
+            val rValueType = data.embedType(variableAssignment.rValue)
+
+            data.withoutResult {
```

Comment:
This doesn't support non-trivial properties, right?  Perhaps there should be a check for them somewhere.

Reply:
The latest commit performs this check!